aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -68,47 +68,63 @@ public class AwsProperties implements Serializable {
    * AwsProperties#S3FILEIO_SSE_TYPE_NONE}.
    *
    * <p>For more details: https://docs.aws.amazon.com/AmazonS3/latest/dev/serv-side-encryption.html
+   *
+   * @deprecated use {@link org.apache.iceberg.aws.s3.S3FileIOProperties} instead
```

Comment:
nit: should also add "will be removed in Iceberg x.x". But I am not sure what's the best version to remove it tbh. @amogh-jahagirdar @nastra any thoughts?

Reply:
Let's first say 1.4 for now, so we at least have those changes available